core/formencoder.go

```diff
@@ -1,12 +1,22 @@
 package core
 
-import "github.com/ggicci/httpin/internal"
+import (
+	"github.com/ggicci/httpin/internal"
+)
 
 type FormEncoder struct {
 	Setter func(key string, value []string) // form value setter
 }
 
 func (e *FormEncoder) Execute(rtm *DirectiveRuntime) error {
+	if rtm.Value.IsZero() {
+		for _, d := range rtm.Resolver.Directives {
+			if d.Name == "omitempty" {
```

Comment:
maybe we could use a constant here to link this back to the actual directive?

Reply:
I think it's okay. But we can use API `rtm.Resolver.GetDirective("omitempty")` here to simplify the code.

Reply:
thanks! missed that.